src/charm_state.py

```diff
@@ -111,12 +110,17 @@ def from_charm(cls, charm: CharmBase) -> "GithubConfig | None":
         path_str = cast(str, charm.config.get(PATH_CONFIG_NAME, ""))
         token = cast(str, charm.config.get(TOKEN_CONFIG_NAME))
 
-        if token == EXPERIMENTAL_JOB_MANAGER_ONLY_TOKEN_VALUE:
-            logger.info("Experimental job manager only token value used.")
-            return None
-
         if not path_str:
             raise CharmConfigInvalidError(f"Missing {PATH_CONFIG_NAME} configuration")
+
+        # check if path_str is an url using pydantic
+        if path_str.startswith("http://") or path_str.startswith("https://"):
+            logger.info(
+                "Detected URL in %s configuration, will use experimental jobmanager mode",
+                PATH_CONFIG_NAME,
+            )
+            return None
```

Comment:
we now use the path for the "hidden" magic to detect the jobmanager mode and use the path for the jobmanager url until we have a proper design